### PR TITLE
Gp 3173 update build output to scratch

### PR DESCRIPTION
**Pull request by Christopher-Ho**
Added an optional output directory argument
- Allows user to choose output directory for test outputs
- This is an optional argument 
- Default behaviour is to output to tmp directory 

Jira ticket: https://jira.oicr.on.ca/browse/GP-3173

- [ X ] Includes a change file
- [ X ] Updates developer documentation

### CODE REVIEW

File: vidarr-cli/src/main/java/ca/on/oicr/gsi/vidarr/cli/TestValidatorScript.java [hunk 1]
```diff
@@ -101,29 +112,35 @@ public boolean validate(String id) {
   }
 
   @JsonProperty("metrics_calculate")
-  public String getMetricsCalculate() {
+  public String getMetricsCalculate()
+  {
```
Comment (avarsava) [inline vidarr-cli/src/main/java/ca/on/oicr/gsi/vidarr/cli/TestValidatorScript.java:116]: Please revert style changes and refer to https://github.com/oicr-gsi/vidarr/blob/master/code-style.md

File: admin-guide.md [hunk 2]
```diff
@@ -270,7 +270,8 @@ validator supported is `"script"`, which runs a calculate script on the output
 files and then runs a comparison script that checks that the calculate script's
 output matches the reference data.
 
-Every provisioned out file will be symlinked into a temporary directory. The
+Users have an option to provide a directory filepath for the provisioned out file. If none is provided then
```
Comment (avarsava) [inline admin-guide.md:273]: Please stick to 80 character lines.

File: changes/add_optional_output_directory_argument.md [hunk 3]
```diff
@@ -0,0 +1,4 @@
+Add an optional output directory argument
```
Comment (avarsava) [inline changes/add_optional_output_directory_argument.md:1]: Please clarify that this is only for vidarr-cli

**Review by avarsava — CHANGES_REQUESTED** [review]
I am confused why these changes couldn't be limited to vidarr-tools? Could vidarr-tools not get the files from the temporary directory and move them to a user-specified location itself?

File: vidarr-pluginapi/src/main/java/ca/on/oicr/gsi/vidarr/ConsumableResource.java [hunk 4]
```diff
@@ -121,12 +123,20 @@ void release(
    * @param workflowName the name of the workflow
    * @param workflowVersion the version of the workflow
    * @param vidarrId the identifier of the workflow run
+   * @param createdTime the time the workflow run was created
```
Comment (grapigeau) [inline vidarr-pluginapi/src/main/java/ca/on/oicr/gsi/vidarr/ConsumableResource.java:126]: Looks like this commit might've been pulled in by accident?

**Comment by avarsava** [comment]
1. Do not revert Andre's changes, that will destroy those changes when merged. Please rebase your branch on master.
2. Please address my question about why this change requires vidarr changes.

**Pull request closed without merging**